### PR TITLE
CIF-2500: Add add-to-wishlist button to PDP and related components

### PR DESCRIPTION
**Pull request by laurentiumagureanu**
## Description

Added add-to-wishlist button to PDP, Product teaser and product recommendation components

## Related Issue

CIF-2500

## Motivation and Context

Implement add to wishlist

## How Has This Been Tested?

unit tests
UI tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes and the overall coverage did not decrease.
- [x] All unit tests pass on CircleCi.
- [x] I ran all tests locally and they pass.

### CODE REVIEW

File: extensions/product-recs/react-components/src/components/ProductRecsGallery/ProductCard.css [hunk 1]
```diff
@@ -30,6 +30,69 @@
...
+    color: white;
+    background-color: rgb(var(--color));
+}
+
+.addToCart:hover {
+    color: white;
+    background-color: rgb(var(--venia-teal));
+}
+
+.addToWishlist {
```
Comment (Buuhuu) [inline extensions/product-recs/react-components/src/components/ProductRecsGallery/ProductCard.css:67]: you can put the redundant styles of `.addToWhishlist and `.addToWhishlist:hover` in a combined css rule like

```
.addToCart:hover,
.addToWishlist:hover {
    color: white;
    background-color: rgb(var(--venia-teal));
}
```

File: react-components/src/components/BundleProductOptions/bundleProductOptions.js [hunk 2]
```diff
@@ -277,6 +297,19 @@ const BundleProductOptions = () => {
...
                 </button>
+                <button
+                    className="button__root_normalPriority button__root clickable__root"
+                    type="button"
+                    onClick={addToWishlist}>
+                    <span className="button__content">
+                        <span>
+                            {intl.formatMessage({
+                                id: 'product:add-item-wishlist',
+                                defaultMessage: 'Add to Wish List'
```
Comment (Buuhuu) [inline react-components/src/components/BundleProductOptions/bundleProductOptions.js:308]: In the productrecs you used `Add to Wishlist` and `productrecs:add-to-wishlist`. Better use `product:add-to-wishlist` here as well. Regarding the label I would keep `Add to Wish List`.

File: extensions/product-recs/react-components/src/components/ProductRecsGallery/ProductCard.js [hunk 3]
```diff
@@ -72,12 +81,17 @@ const ProductCard = props => {
...
+                <Trigger className={classes.buttonMargin} action={() => addToCart(props.unit, props.product)}>
                     <span className={classes.addToCart}>
-                        {intl.formatMessage({ id: 'productrecs:add-to-cart', defaultMessage: 'Add to cart' })}
+                        {intl.formatMessage({ id: 'productrecs:add-to-cart', defaultMessage: 'Add to Cart' })}
                     </span>
                 </Trigger>
             )}
+            <Trigger className={classes.buttonMargin} action={() => addToWishlist(props.product)}>
+                <span className={classes.addToWishlist}>
+                    {intl.formatMessage({ id: 'productrecs:add-to-wishlist', defaultMessage: 'Add to Wishlist' })}
```
Comment (Buuhuu) [inline extensions/product-recs/react-components/src/components/ProductRecsGallery/ProductCard.js:92]: it is `Wish List`

**Review by Buuhuu — APPROVED** [review]

**Review by Buuhuu — COMMENTED** [review]
I think we should introduce a policy configuration to allow editors to hide the wish list for shops that don't need/want it. Should be disabled per default.

**Review by herzog31 — APPROVED** [review]

**Pull request merged**